### PR TITLE
Add "getIdToken" user method

### PR DESCRIPTION
**Pull request by filwiesner**
Add method for retrieving ID token used for server side authentication

### CODE REVIEW

File: src/jsMain/kotlin/dev/bitspittle/firebase/externals/auth/Externals.kt [hunk 1]
```diff
@@ -74,6 +74,9 @@ internal external interface User : UserInfo {
     val metadata: UserMetadata
     val providerData: Array<UserInfo>
     val refreshToken: String
+
+    // https://firebase.google.com/docs/reference/js/v8/firebase.User#getidtoken
```
Comment (bitspittle) [inline src/jsMain/kotlin/dev/bitspittle/firebase/externals/auth/Externals.kt]: Nit: Let's link to the auth version of the docs, for consistency with the rest of the file

https://firebase.google.com/docs/reference/js/auth.user#usergetidtoken

File: src/jsMain/kotlin/dev/bitspittle/firebase/auth/Bindings.kt [hunk 2]
```diff
@@ -118,6 +118,12 @@ class User internal constructor(
 
     suspend fun sendEmailVerification() =
         dev.bitspittle.firebase.externals.auth.sendEmailVerification(wrapped).await()
+
+    /**
+     * Returns a JSON Web Token (JWT) used to identify the user to a Firebase service.
+     * Returns the current token if it has not expired. Otherwise, this will refresh the token and return a new one.
+     */
+    suspend fun getIdToken(forceRefresh: Boolean? = null) = wrapped.getIdToken(forceRefresh).await()
```
Comment (bitspittle) [inline src/jsMain/kotlin/dev/bitspittle/firebase/auth/Bindings.kt:123]: Nit: Would you mind alphabetizing this method within this class?

It's not a big deal, really -- I just tend to either order methods intentionally (if there's a clear grouping) or, without that, alphabetize them.

The reason is, as a general strategy, it reduces the chance of merge conflicts if a project ever grows to the point of multiple people working on it. (If everyone just adds fields / methods to the end, then merges will inevitably start stomping on each other)

**Review by bitspittle — CHANGES_REQUESTED** [review]
Thank you so much for the PR!

Some comments here, that should be trivial to resolve.

BTW, if my comments are ever long, it's not because I'm pushing back forcefully -- I just want to make sure my thoughts are clear so that you could push back on the push back if you wanted to.

Grateful for the contribution!

File: src/jsMain/kotlin/dev/bitspittle/firebase/auth/Bindings.kt (hunk 2 above)
Comment (wooodenleg) [reply src/jsMain/kotlin/dev/bitspittle/firebase/auth/Bindings.kt:123]: Fair enough 👍

Comment (filwiesner) [reply src/jsMain/kotlin/dev/bitspittle/firebase/auth/Bindings.kt:123]: Fair enough 👍

**Comment by filwiesner** [comment]
> BTW, if my comments are ever long, it's not because I'm pushing back forcefully -- I just want to make sure my thoughts are clear so that you could push back on the push back if you wanted to.

I don't mind :D I am really pedantic in my code reviews as well so I actually like when someone takes time to give proper criticism

**Review by bitspittle — APPROVED** [review]
Thanks again!

**Pull request merged**